src/include/mpir_datatype.h

```diff
@@ -492,6 +492,43 @@ static inline int MPIR_Datatype_set_contents(MPIR_Datatype * new_dtp,
     return MPI_SUCCESS;
 }
 
+MPL_STATIC_INLINE_PREFIX void MPIR_Datatype_access_contents(MPIR_Datatype_contents * cp,
```

Comment:
Suggest `MPIR_Datatype_get_contents`.

Reply:
Nevermind, I see later in the history what is going on. Naming is still weird to me, but not your change.

Reply:
There is `MPI_Type_get_contents`, which return *copies* of the internal counts array. `MPIR_Datatype_access_contents` just returns the pointer to the internal data, I think the name distinction is important.